provisio-core/src/main/java/ca/vanzyl/provisio/action/artifact/WriteToDiskAction.java

```diff
@@ -41,20 +46,33 @@ public WriteToDiskAction(ProvisioArtifact artifact, File outputDirectory) {
 
     @Override
     public void execute(ProvisioningContext context) {
-        File file = artifact.getFile();
-        if (file != null) {
-            copy(file, new File(outputDirectory, targetArtifactFileName(artifact, file.getName())));
+        if (artifact.getFile() != null) {
+            write(
+                    context,
+                    artifact,
+                    targetArtifactFileName(context, artifact, artifact.getFile().getName()));
         }
     }
 
-    public void copy(File source, File target) {
+    private void write(ProvisioningContext context, ProvisioArtifact source, String targetPath) {
+        File target = new File(outputDirectory, targetPath);
         try {
-            if (!target.getParentFile().exists()) {
-                target.getParentFile().mkdirs();
+            Files.createDirectories(target.getParentFile().toPath());
+            if (ProvisioVariables.allowTargetOverwrite(context)) {
+                if (Files.isRegularFile(target.toPath())) {
```

Comment:
What happens if you rebuild without having cleaned ?

Reply:
Good question... Should I maybe use that change detecting stream from p-u?

Reply:
That would be very nice ! Ideally, the assemblies would not be rebuilt if nothing changed, but that may be a different thing.

Reply:
Woah, that would be a killer feature

Reply:
problem is that provision is "usually" open up something, mangle, archive up something. UnArchiver now will not overwrite if files present, nor will WriteToDiskAction, but last element is Archiver (that am unsure how could we fix)